confidence/formats.py

```diff
@@ -0,0 +1,78 @@
+import json
+import typing
+from abc import abstractmethod
+from dataclasses import dataclass, replace
+from os import PathLike
+from pathlib import Path
+
+import yaml
+
+from confidence.models import unwrap
+
+
+@dataclass(frozen=True)
+class Format(typing.Protocol):
+    suffix: str = ''
+    encoding: str = 'utf-8'
```

Comment:
Could be a literal >> https://github.com/python/cpython/blob/3.13/Lib/encodings/aliases.py

Reply:
Ah interesting, you're right that not any encoding will just work, and that link is likely the source of the values that are going to work. At the moment not yet sure how to refer to that typing-wise (instead of the `str` here), but let's try 👌🏻 

I'll take a look at how the builtins like `open()` type this.

Reply:
Hmm, unfortunately, the builtin `open()`, as defined by the typeshed, doesn't seem to tackle this very smartly, it just states `str | None`… As explicit is better than implicit, I'm leaning towards leaving it as `str` here.

https://github.com/python/typeshed/blob/main/stdlib/builtins.pyi